app/controllers/oauth_clients_controller.rb

```diff
@@ -19,7 +19,11 @@ def show
   end
 
   def new
-    @client_application = ClientApplication.new
+    if Settings.oauth_10_registration
+      @client_application = ClientApplication.new
+    else
+      redirect_to :action => "index"
```

Comment:
I think this needs a flash message. If we're doing more than just removing the link from the index page, then I guess we're trying to handle external links to `:new` e.g. from a tutorial somewhere. A flash message would explain why the link to `:new` no longer works.

Reply:
Does `:create` need a similar redirect? Or are we happy that if the form is removed, then anyone going to the lengths of making a direct `:post` with the correct CSRF token is only causing problems for themselves?

Reply:
I've added the flash message - as you say I didn't bother with `:create` because you'd have to try really hard and this is only going to be short lived.